Stripe/STPCheckoutAPIClient.m

```diff
@@ -77,27 +78,28 @@ - (BOOL)readyForLookups {
 }
 
 - (STPPromise *)lookupEmail:(NSString *)email {
-    __weak typeof(self) weakself = self;
+    WEAK(self);
+    Class selfClass = self.class;
```

Comment:
So if you pull out selfClass here, then you are capturing `selfClass`. If you reference `self.class` later, that is capturing `self` (and being converted to `[self class]` basically). Under the previous code, the weak self can never actually be nil because you are capturing it inside the `if (!weakself) {}` block. In the new method, the if statement works correctly, plus we still have a ref to the class type for use in the error message.